### PR TITLE
Add github style links to docs sections

### LINKED ISSUE #5106: Add support for a :link: symbol on hover for docs headers?
[Zulip discussion](https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/docs.20improvements/near/339841055) 

> perhaps make a # before the function name for header linking? may not look as clean but much more ergonomic for sure
> How about a :link: symbol on hover like github does for markdown headers?

### PR DESCRIPTION
**Pull request by itmuckel**
Fixes #5106.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -106,6 +106,7 @@
           cargo-criterion # for benchmarks
           simple-http-server # to view roc website when trying out edits
           wasm-pack # for repl_wasm
+          jq
```
Comment (itmuckel) [inline flake.nix:109]: This was required for `./www/build.sh` to work.

**Review by Anton-4 — CHANGES_REQUESTED** [review]
Thanks for your PR @itmuckel!
There are some small things that can be improved:
- Could the link "icon" be changed to have the same color as the type next to it? Right now it might not contrast enough with the background in dark mode for people who require a higher contrast to see well.
- Double-clicking the type and pressing Ctrl+C now also copies the link icon. Can you see if that can be avoided?
- Would it be possible to de-activate the hover after say 3 seconds instead of instantly when the mouse moves? Otherwise you have to be sort of careful with your mouse movement to make the link icon not disappear. This is not a big deal for most people but could be really annoying for those with a disability.

This is no list of tasks you must do, I'm happy to take care of it if you can't find the time.

**Comment by Anton-4** [comment]
The next few days is fine, on firefox there are [some buttons](https://discourse.mozilla.org/t/how-to-test-preferred-color-scheme-dark-mode-with-firefox/40769/5) in the developer console (Ctrl+Shift+i).

**Comment by itmuckel** [comment]
@Anton-4 Okay, that should be it 🙂

**Comment by itmuckel** [comment]
Sorry, I've seen that the commits weren't linked to my account and amended them with my correct email. 😅

**Review by Anton-4 — APPROVED** [review]

**Comment by Anton-4** [comment]
Can you merge this @rtfeldman?

**Comment by itmuckel** [comment]
Rebased and signed commits. 🙂

**Review by Anton-4 — APPROVED** [review]
Thanks @itmuckel!

**Pull request merged**